aa/pb.py

```diff
@@ -295,13 +296,79 @@ class PbFileFetcher(fetcher.Fetcher):
     def __init__(self, root):
         self._root = root
 
-    def _get_pb_file(self, pv, year):
-        # Split PV on either dash or colon
+    def _get_all_pb_files_of_pv(self, pv):
+        """Returns a list of all .pb files of the given PV which are found under the
+        root directory. Assuming the storage is split into STS/MTS/LTS."""
+        # Strip any STS/MTS/LTS from the root first to ensure backwards compatibility.
+        if os.path.basename(os.path.normpath(self._root)) in ["STS", "MTS", "LTS"]:
+            self._root = os.path.dirname(os.path.normpath(self._root))
+        pv_files = []
+        # Split PV on either dash or colon.
         parts = re.split("[-:]", pv)
         suffix = parts.pop()
-        directory = os.path.join(self._root, os.path.sep.join(parts))
-        filename = "{}:{}.pb".format(suffix, year)
-        return os.path.join(directory, filename)
+        # Go through long/mid/short term storages and collect all files for this PV.
+        for s in ["LTS", "MTS", "STS"]:
+            directory = os.path.join(self._root, s, os.path.sep.join(parts))
+            for f in sorted(glob.glob(os.path.join(directory, f"{suffix}*pb"))):
+                pv_files.append(f)
+        return pv_files
+
+    def _create_datetime_for_pb_file(self, filepath):
```

Comment:
This could be a static method or just a function at module level, because it doesn't use any knowledge of the `PbFileFetcher` object. This may make it easier to test.